### PR TITLE
XML injection

### PR DESCRIPTION
**Pull request by BlueSquare1**

### CODE REVIEW

File: MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"risk_rating": "high", "short_description": "XML Injection is a type of attack where an attacker injects malicious XML code into an application, which can then be executed by the application.", "references": {"External Entity Injection (OWASP)": "https://owasp.org/www-community/vulnerabilities/XML_External_Entity_(XXE)_Processing", "XPath Injection (OWASP)": "https://owasp.org/www-community/attacks/XPATH_Injection"}, "title": "XML Injection", "privacy_issue": false, "security_issue": true, "categories": {}}
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json:1]: Please reformat this file add missing categories.

Comment (3asm) [inline MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json:1]: The description is not correct. Executed by the application?!

File: MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/description.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+## External Entity Injection
+
+External Entity Injection is a type of XML Injection where an attacker injects an external entity into an XML document. This can be used to read sensitive data, perform denial of service attacks, or execute remote code.
+
+### Examples
+
+
+#### Swift
+
+```{language}
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/description.md:14]: These are not formatted correctly. Please recheck your PR content and code before we do the review.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json (hunk 1 above)
Comment (BlueSquare1) [reply MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json:1]: Well isn't the point of those PRs to demonstrate how well AI can perform in KB generation with minimal interference from our side? if I fix everything before sending my PRs you wouldn't know if it was the AI that generated that content or me who added it manually.

Comment (3asm) [reply MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json:1]: No, absolutely not.

File: MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+	"risk_rating": "high",
+	"short_description": "XML Injection is a type of attack where an attacker injects malicious XML code into an application, which can then be executed by the application.",
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/meta.json:3]: Executed by the application is still left.

File: MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/description.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
+# XML Injection
+
+XML injection is a security vulnerability that occurs when user-supplied input in an application is not properly validated and is included in dynamically generated XML documents. This vulnerability allows an attacker to manipulate the structure or content of the XML document, leading to various malicious activities. By injecting malicious XML code, an attacker can execute arbitrary commands, read sensitive data, or modify the behavior of the application.
+
+## External Entity Injection
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_MEDIUM/XML_INJECTION/description.md:5]: XXE, XML Injection, XPath injection will have different KBs.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**